File: src/code-samples/langchain/long-term-memory-create-agent-postgres.ts
// :snippet-start: long-term-memory-create-agent-postgres-js
import { createAgent } from "langchain";
import { PostgresStore } from "@langchain/langgraph-checkpoint-postgres/store";

const DB_URI =
  process.env.POSTGRES_URI ??
  "postgresql://postgres:postgres@localhost:5432/postgres?sslmode=disable";
const store = PostgresStore.fromConnString(DB_URI);
await store.setup();

const agent = createAgent({
  model: "claude-sonnet-4-6",
  tools: [],
  store,
});
// :snippet-end:

// :remove-start:
if (!agent) {
  throw new Error("Agent creation failed");
}
console.log("✓ Agent with PostgresStore created successfully");
await store.stop();
// :remove-end:

File: src/code-samples/langchain/long-term-memory-create-agent-postgres.py
# :snippet-start: long-term-memory-create-agent-postgres-py
from langchain.agents import create_agent
from langchain_core.runnables import Runnable
from langgraph.store.postgres import PostgresStore  # type: ignore[import-not-found]

DB_URI = "postgresql://postgres:postgres@localhost:5432/postgres?sslmode=disable"
# :remove-start:
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parent.parent))
from conftest import get_postgres_uri, prepare_postgres_store

DB_URI = get_postgres_uri()
prepare_postgres_store(DB_URI)
# :remove-end:

with PostgresStore.from_conn_string(DB_URI) as store:
    store.setup()
    agent: Runnable = create_agent(
        "claude-sonnet-4-6",
        tools=[],
        store=store,
    )
# :snippet-end:

# :remove-start:
assert agent is not None
print("✓ Agent with PostgresStore created successfully")
# :remove-end:
